cubed/core/ops.py

```diff
@@ -729,24 +729,20 @@ def reduction(
     allowed_mem = x.spec.allowed_mem
     max_mem = allowed_mem - x.spec.reserved_mem
 
-    # reduce initial chunks (if any axis chunksize is > 1)
-    if (
-        any(s > 1 for i, s in enumerate(result.chunksize) if i in axis)
-        or func != combine_func
-    ):
```

Comment:
This condition needs to be removed since otherwise there are cases where no function is ever run in `reduction`. In particular if the array is of size 1 along the reduction axis. So far this hasn't been a problem since the reduction of a single element has always been that element (e.g. `sum(1)` is 1). But that doesn't hold for NaN (e.g. `nansum(nan)` is 0).